library/src/test/java/com/okta/appauth/android/OktaAppAuthTest.java

```diff
@@ -176,6 +173,97 @@ public void testLogoutNoLogedinUserFoundException(){
+                .thenReturn(new CustomTabsIntent.Builder());
+        when(mAuthStateManager.getCurrent()).thenReturn(authState);
+        when(authState.getAuthorizationServiceConfiguration())
+                .thenReturn(TestUtils.getTestServiceConfig());
+        sut.mAuthRequest.set(TestUtils.getTestAuthRequest());
+
+        sut.login(mContext, success, failure);
+
+        ArgumentCaptor<AuthorizationRequest> argument = ArgumentCaptor.forClass(AuthorizationRequest.class);
+        verify(mAuthService, times(1))
+                .performAuthorizationRequest(
+                        argument.capture()
+                        ,any(PendingIntent.class)
+                        ,any(PendingIntent.class)
+                        ,any(CustomTabsIntent.class)
+                );
+
+        assertThat(expectedRequest.clientId)
+                .isEqualTo(argument.getValue().clientId);
+        assertThat(expectedRequest.configuration.toJsonString())
+                .isEqualTo(argument.getValue().configuration.toJsonString());
+    }
+
+    @Test
+    public void testLoginWithPayloadSuccess() {
+        PendingIntent success = mock(PendingIntent.class);
+        PendingIntent failure = mock(PendingIntent.class);
+        AuthState authState = mock(AuthState.class);
+
+        when(mAuthService.createCustomTabsIntentBuilder(any(Uri.class)))
+                .thenReturn(new CustomTabsIntent.Builder());
+        when(mAuthStateManager.getCurrent()).thenReturn(authState);
+        when(authState.getAuthorizationServiceConfiguration())
+                .thenReturn(TestUtils.getTestServiceConfig());
+        when(mConfiguration.getRedirectUri()).thenReturn(TestUtils.TEST_APP_REDIRECT_URI);
+        when(mConfiguration.getScopes()).thenReturn(new HashSet<>(TestUtils.TEST_SCOPES_SUPPORTED));
+        sut.mClientId.set(TestUtils.TEST_CLIENT_ID);
+
+        AuthPayload payload = new AuthPayload.Builder()
+                .addParameter("testName", "testValue")
+                .setSteate("testState")
+                .setmLoginHint("loginHint")
+                .build();
+
+        sut.login(mContext, success, failure, payload);
+
+        ArgumentCaptor<AuthorizationRequest> argument = ArgumentCaptor
+                .forClass(AuthorizationRequest.class);
+
+        verify(mAuthService, times(1))
+                .performAuthorizationRequest(
+                        argument.capture()
+                        ,any(PendingIntent.class)
+                        ,any(PendingIntent.class)
+                        ,any(CustomTabsIntent.class)
+                );
+
+        assertThat(argument.getValue().loginHint)
+                .isEqualTo(payload.getLoginHint());
+        assertThat(argument.getValue().additionalParameters)
+                .isEqualTo(payload.getAdditionalParameters());
+        assertThat(argument.getValue().state)
+                .isEqualTo(payload.getState());
+    }
+
+    @Test(expected = IllegalStateException.class)
```

Comment:
nit: mockito could also throw an IllegalStateException, which could provide a false positive pass for this test.
I think Hamcrest and AssertJ both have ways to assert exceptions, and of course there is the old school try/catch option (or equivalent lambda)

Reply:
Our AssertJ does not support exception handling, but I used try-catch for that, should fix false positive issue.

Reply:
Sorry, I usually test with groovy and use a closure for this stuff, my memory is a little rusty 😄